asr1k_neutron_l3/plugins/l3/schedulers/simple_asr1k_scheduler.py

```diff
@@ -43,12 +45,27 @@ def _get_candidates(self, plugin, context, sync_router):
                            'agent_id': current_l3_agents[0]['id']})
                 return []
 
-            candidates = plugin.get_l3_agents(context, active=True)
+            orig_candidates = plugin.get_l3_agents(context, active=True)
+
+            # router creation with az hint: only schedule on agent with appropriate AZ
+            # router creation without az hint: only schedule on agent with no AZ
+            az_hints = orig_az_hints = self._get_az_hints(sync_router)
+            if not az_hints or az_hints[0] in asr1k_const.NO_AZ_LIST:
+                az_hints = asr1k_const.NO_AZ_LIST
+            candidates = [c for c in orig_candidates if c.availability_zone in az_hints]
+
+            if not candidates and orig_az_hints and cfg.CONF.asr1k.ignore_invalid_az_hint_for_router:
```

Comment:
`_get_az_hints` returns either the az or the default config option. So bool(`orig_az_hints`) will always be true, right?

Reply:
This is the implementation of `_get_az_hints()`: https://github.com/sapcc/neutron/blob/d6ba5ad3c418fa5ce99c7317966b66d0c525be4e/neutron/scheduler/l3_agent_scheduler.py#L368-L370

so it either returns the router's at hints or the default az, which in our infra is empty:
```
>>> from oslo_config import cfg
>>> cfg.CONF.default_availability_zones
[]
```
`bool([])` would then be False.

Reply:
For some reason i thought that setting defaults to 'nova'.